src/oemof/solph/flows/_flow.py

```diff
@@ -220,6 +219,24 @@ def __init__(self, **kwargs):
                 + "nonconvex flows!"
             )
 
+        if not self.investment:
+            warn_msg = (
+                "If {} is set in a dispatch model, "
```

Comment:
```suggestion
                "If {} is set in a flow (except InvestmentFlow), "
```
I mean you could also think of "mixed" models (but thats really a small thing ;-)